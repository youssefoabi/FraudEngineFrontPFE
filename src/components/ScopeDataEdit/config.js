// eslint-disable-next-line import/prefer-default-export
export const config = [
  {
    columns: 4,
    id: 'name',
    type: 'input',
    label: 'Nom de la gamme',
  },
  {
    columns: 12,
    id: 'urlTemplate',
    type: 'input',
    label: 'Url avec param√®tres',
  },
];
