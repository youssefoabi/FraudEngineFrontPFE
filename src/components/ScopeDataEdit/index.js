import { DataEdit, GridItem } from '@cdiscount/backoffice-ui';
import { func, number, shape, string } from 'prop-types';
import React from 'react';

import { config } from './config';

const ScopeDataEdit = ({ scope, handleChange }) => (
  <GridItem title="CaractÃ©ristiques de la gamme">
    <DataEdit fields={config} values={scope} onChange={handleChange} />
  </GridItem>
);

export const scopeShape = shape({
  id: number,
  modifiedAt: number,
  name: string,
  urlTemplate: string,
});

ScopeDataEdit.propTypes = {
  handleChange: func,
  scope: scopeShape,
};

ScopeDataEdit.defaultProps = {
  handleChange: Function.prototype,
  scope: null,
};

export default ScopeDataEdit;
