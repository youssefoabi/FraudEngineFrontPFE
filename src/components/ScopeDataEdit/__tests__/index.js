import { itRendersCorrectly } from '@cdiscount/jest-utils';
import React from 'react';

import ScopeDataEdit from '../';

describe(ScopeDataEdit.name, () => {
  itRendersCorrectly(<ScopeDataEdit />);
});
