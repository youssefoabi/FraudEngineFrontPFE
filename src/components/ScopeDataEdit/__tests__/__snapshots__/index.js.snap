// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScopeDataEdit renders correctly 1`] = `
<h
  fullheight={false}
  title="Caractéristiques de la gamme"
>
  <f
    config={null}
    fields={
      Array [
        Object {
          "columns": 4,
          "id": "name",
          "label": "Nom de la gamme",
          "type": "input",
        },
        Object {
          "columns": 12,
          "id": "urlTemplate",
          "label": "Url avec paramètres",
          "type": "input",
        },
      ]
    }
    inputs={null}
    onChange={[Function]}
    onInit={[Function]}
    title={null}
    values={null}
  />
</h>
`;
